onnxruntime/core/optimizer/constant_sharing.cc

```diff
@@ -29,20 +29,35 @@ bool IsSupportedDataType(int32_t data_type) {
 
 using SupportedTypeList = boost::mp11::mp_list<MLFloat16, float, double, int32_t, int64_t>;
 
-bool IsValidSingleValueShape(const ONNX_NAMESPACE::TensorShapeProto* input_shape) {
+static constexpr int64_t MAX_SIZE_PER_VALUE = 8;
```

Comment:
why do we have this restriction on num of elements?

Reply:
Having a bigger tensor element size threshold here means more overhead running ConstantSharing graph transformation. Originally the number is 1, now I changed it to 8 gradually. Maybe we can make it bigger once we found it helps for some specific scenarios.

Reply:
Can you add some cooments around the reasoning behind choosing 8 and what should one consider if they want to change this or remove this limitation altogether in future. Thanks!

Reply:
Sure, I added a comment for it. :)